c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -48,16 +49,25 @@ impl<'tcx> TypeChecker<'tcx, '_> {
     }
 
     pub fn visit_place_ref(&mut self, pl: PlaceRef<'tcx>, ctx: PlaceContext) -> LTy<'tcx> {
+        let mutbl = match ctx {
```

Comment:
do you think this would be useful as its own utility function?

Reply:
It turns out the use of `PlaceContext` was a holdover from previous code that was implemented as a `mir::Visitor`.  I changed these functions to take `mutbl: Mutability` directly instead.